aten/src/ATen/native/cpu/SpmmReduceKernel.cpp

```diff
@@ -0,0 +1,601 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
```

Comment:
Naive question: Can't we share code for this from normal `spmm`?

Reply:
`spmm` from mkl have several limitations:
* it only applies to reduce type of `sum`
* do not have reduced dtypes, e.g. `BFloat16` or `half`
* index needs to be `int32_t`, by default the indices from torch is `int64_t` which means an extra dtype conversion is needed.
* values need to be contiguous. The SparseCSR tensor in torch now supports non-contiguous values which means we can do broadcast on values. but this won't apply to spmm.

Anyway I have made some compare that the kernel here is not worse than spmm.